### PR TITLE
pool_workflows: object and snapshot creation/deletion

### PR DESCRIPTION
**Pull request by pdhiran**
Signed-off-by: Pawan Dhiran <pdhiran@redhat.com>

# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [x] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [x] Implement the test script and perform test runs
- [x] Submit PR for code review and approve
- [x] Update Polarin Test with Automation script details and update automation fields
- [x] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

**Review by tintumathew10 — COMMENTED** [review]
LGTM

File: ceph/rados/pool_workflows.py [hunk 1]
```diff
@@ -123,3 +123,130 @@ def fill_omap_entries(self, pool_name, **kwargs):
...
+        Args:
+            pool_name: name of the pool
+        Returns: Pass -> name of the snapshot created, Fail -> False
+
+        """
+        # Checking if snapshots can be created on the supplied pool
+        cmd = "ceph osd dump"
+        pool_status = self.rados_obj.run_ceph_command(cmd=cmd)
+        for detail in pool_status["pools"]:
+            if detail["pool_name"] == pool_name:
```
Comment (psathyan) [inline ceph/rados/pool_workflows.py:176]: You could do 
```
if detail["pool_name"] != pool_name:
  continue
```

File: ceph/rados/pool_workflows.py [hunk 2]
```diff
@@ -123,3 +123,130 @@ def fill_omap_entries(self, pool_name, **kwargs):
             f"Total stored omap data on pool : {total_omap_data}"
         )
         return True
+
+    def do_rados_delete(self, **kwargs):
```
[the remark this answers is not in the document]
Comment (pdhiran) [reply ceph/rados/pool_workflows.py:127]: We already have methods named ` do_rados_put ` and ` do_rados_get ` , which put and get the objects from the pool resp. To maintain similar naming conventions, named the method ` do_rados_delete `, which deletes the objects from the pool.

File: ceph/rados/pool_workflows.py [hunk 3]
```diff
@@ -123,3 +123,130 @@ def fill_omap_entries(self, pool_name, **kwargs):
...
+            Allowed values :
+                1. pool_name: name of the pool
+                2. pg_id: Pg ID ( to be provided along with pool name )
+
+        Returns: True -> pass, False -> fail
+        """
+        try:
+            pool_name = kwargs["pool_name"]
+        except KeyError:
+            log.error("Pool name not provided to delete objects")
```
[the remark this answers is not in the document]
Comment (pdhiran) [reply ceph/rados/pool_workflows.py:141]: Fixed the docstring. Yes, the PG_ID was a optional param. what was intended is ` [ pg_id ]: Pg ID (Optional, but when provided, should be passed along with pool name ) `

Pool name is a compulsory arg that needs to be passed. 
Changed the method from accepting **kwargs to accepting named params.

File: ceph/rados/pool_workflows.py (hunk 1 above)
Comment (pdhiran) [reply ceph/rados/pool_workflows.py:176]: Updated.

File: ceph/rados/pool_workflows.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (pdhiran) [reply ceph/rados/pool_workflows.py]: Also, we can pass pg_id, where only objects in that particular PG would be deleted, not entire pool. So does not seem right naming ` delete_objects_in_pool ` .

**Comment by pdhiran** [comment]
rebased with latest code changes

**Review by psathyan — APPROVED** [review]

**Pull request merged**